### PR TITLE
docs: correct type links, add admonitions, support extends annotation

### PR DESCRIPTION
**Pull request by lukewalczak**
Fixes: https://github.com/callstack/react-native-paper/issues/3731

### Summary

Addressing leftovers from the list within the [issue](#3731):

* corrects link to the egghead on the contribution guide
* adding admonitions in several places: 

image

* add support for `@extends` annotation, for prop extension:

image

* correct links for `Edit this page` in `guides`
* correct links for types such as `IconSource` and `ThemeProp`

### CODE REVIEW

File: docs/component-docs-plugin/generatePageMDX.js [hunk 1]
```diff
@@ -34,7 +34,8 @@ function generatePageMDX(doc, link) {
   const description = doc.description
     .replace(/<\/br>/g, '')
     .replace(/style="[a-zA-Z0-9:;.\s()\-,]*"/gi, '')
-    .replace(/src="screenshots/g, `src="${baseUrl}screenshots`);
+    .replace(/src="screenshots/g, `src="${baseUrl}screenshots`)
+    .replace(/@extends.+$/, '');
```
Comment (lukewalczak) [inline docs/component-docs-plugin/generatePageMDX.js:38]: Removing from the description the part related to extending other prop e.g. "@extends TouchableRipple Props"

File: docs/src/components/PropTable.tsx [hunk 2]
```diff
@@ -25,6 +28,38 @@ const renderBadge = (annotation: string) => {
   return `<span class="badge badge-${annotType.replace('@', '')} ">${annotLabel.join(' ')}</span>`;
 };
 
+const renderExtendsLink = (description: string) => {
```
Comment (lukewalczak) [inline docs/src/components/PropTable.tsx:31]: Adding support extending prop for "@extends ....."

**Comment by hjonasson** [comment]
Hey @lukewalczak . Does this PR solve that in these docs [here](https://callstack.github.io/react-native-paper/docs/components/List/ListIcon) the `IconProps` link doesn't work? If so I will not submit my own ticket (and I couldn't find where it was supposed to link)

**Comment by lukewalczak** [comment]
Hey @hjonasson, within the PR, there is a fix for that and `IconSource` pointing to the [type](https://github.com/callstack/react-native-paper/blob/main/src/components/Icon.tsx#L16) in the source code.

**Pull request merged**